static/js/components/CommentTree.js

```diff
@@ -125,21 +124,13 @@ export default class CommentTree extends React.Component<Props> {
             downvote={downvote}
           />
         ) : null}
-        {atMaxDepth ||
-        moderationUI ||
-        comment.deleted ||
-        userIsAnonymous() ? null : (
-            <div
-              className="comment-action-button reply-button"
-              onClick={e => {
-                if (beginEditing) {
-                  beginEditing(formKey, initialValue, e)
-                }
-              }}
-            >
-            reply
-            </div>
-          )}
+        {atMaxDepth || moderationUI || comment.deleted ? null : (
+          <ReplyButton
+            beginEditing={beginEditing}
+            formKey={replyToCommentKey(comment)}
+            initialValue={getCommentReplyInitialValue(comment)}
```

Comment:
what do you think about just passing down a `beginEditing` func to `ReplyButton`? It seems like the `ReplyButton` shouldn't have to know about `formKey` and `initialValue` and all that

Reply:
sure